crates/matrix-sdk/src/client/mod.rs

```diff
@@ -409,13 +425,22 @@ impl Client {
     /// The authentication server info discovered from the homeserver.
     ///
     /// This will only be set if the homeserver supports authenticating via
-    /// OpenID Connect ([MSC3861]) and this `Client` was constructed using
-    /// auto-discovery by setting the homeserver with
-    /// [`ClientBuilder::server_name()`].
+    /// OpenID Connect ([MSC3861]) and either this `Client` was constructed
+    /// using auto-discovery by setting the homeserver with
+    /// [`ClientBuilder::server_name()`] or it is authenticated using said
+    /// authentication server.
     ///
     /// [MSC3861]: https://github.com/matrix-org/matrix-spec-proposals/pull/3861
     pub fn authentication_server_info(&self) -> Option<&AuthenticationServerInfo> {
-        self.inner.authentication_server_info.as_ref()
+        if let Some(discovered_server) = &self.inner.authentication_server_info {
```

Comment:
It is on purpose that it only returns something if the homeserver was freshly discovered. The issuer and account management URL can be retrieved from the Oidc API after login.

Reply:
Reverting 👍